book/src/async_support.md

```diff
@@ -0,0 +1,40 @@
+# Async Executors
+
+Ariel OS embraces async Rust and favors async interfaces over blocking ones.
+
+## System Executor Flavors
+
+Ariel OS provides a system executor which is automatically started at startup.
+It comes in three different flavors, identified by their laze module names:
+
+| laze module              | Embassy executor | Description |
+| ------------------------ | ---------------- | ----------- |
+| `executor-interrupt`     | [InterruptExecutor][interrupt-executor-rustdoc] | Runs in handler mode. A software interrupt (SWI) handler is used when the MCU provides one, otherwise the [board configuration must specify](./adding_board_support.md#adding-support-for-a-board-1) which peripheral interrupt to dedicate to this executor. |
+| `executor-single-thread` | [Executor][executor-rustdoc] | Runs in thread mode in place of threads. Therefore not compatible with [multithreading][multithreading-book]. |
+| `executor-thread`        | [Custom, based on `raw::Executor`][asynch-thread-executor-rustdoc] | Runs inside a dedicated thread automatically started at startup. |
+
+A default flavor compatible with the MCU is automatically selected by default in the order of preference in which they are listed above.
+Another flavor can be manually selected, replacing the default one, by [selecting its laze module][laze-modules-book].
+Not all flavors are available on all MCUs however, and the laze configuration will only allow selecting one the compatible ones.
+
```

Comment:
Indicate that enabling threads will force `executor-thread`?

Reply:
And maybe, that the interrupt executor has a slight theoretical performance advantage, which should be negligible in practice?

Reply:
Hm. The simplest way of quantifying this would be to connect two gpio, await a gpio change in one task, take `now()` & set the other gpio in another task, take `now()` again in the woken up gpio waiter and show the difference. That should be faster in the isr executor, in theory.

... there is also the fact that some sync constructs starve (lower-prio) threads when they degrade to busy loops.

Reply:
thanks for measuring!